HIP/hip-0000-Transaction-Signing-Protocol.md

```diff
@@ -0,0 +1,163 @@
+- hip: XX
+- title: Hedera Signing Protocol
+- author: [0xJepsen](https://github.com/0xJepsen), [rocketmay](https://github.com/rocketmay), [bugBytes](https://github.com/bugbytesinc)
+- type: Standards Track
+- category: Application
+- status: Draft 
+- created: <07/03/21>
+- discussions-to: [here](https://github.com/hashgraph/hedera-improvement-proposal/discussions/98)
+- updated: <07/12/21> 
+- requires: <HIP number(s)>
+- replaces: <HIP number(s)>
+- superseded-by: <HIP number(s)>
+
+## Abstract
+
+This specification proposes a protocol for exchanging unsigned transactions and signatures.
+
+
+## Motivation
+
+Developers are currently creating web authentication mechanisms from scratch for Hedera-based web apps. This limits consumer adoption and results in a poor user and developer experience. A standard protocol for decentralized applications to communicate with clients and allow clients to sign transactions would significantly improve the developer and user experience. The Hedera development space is still young, with many new projects developing promising ideas. These projects will need a standardized method of communication, much like what exists in the Ethereum space.
+
+Taking the Metamask wallet as an example, Metamask offers a seamless way for users to interact with dApps, sending requests and signing transactions without compromising the security of their accounts. Another example is Wallet Connect, which is an open protocol for signing ETH transactions.
+
+Without a communication standard, projects in the space are required to reinvent the wheel for every application and wallet. This adds overhead for the project team. This also represents a significant risk to the user who must reveal sensitive account information to web applications in order to interact with the Hedera network.
+
+## Rationale
+
+We propose establishing a standard protocol for sending and receiving transactions to be signed. There is no sensitive account information within the transaction data. We prepose protobufs for four RPC calls and their corresponding requests and responses. The four calls shall be `RequestLimit`, `Sign`, `Submit`, and `SubmitAndSign`.  
+
+* `RequestLimit` allows for information like `MaxTransactionFee` and `PreAuthorizationLimit` to be obtained entities utilizing the protocol. 
+* `Sign` allows an entity to provide a `SigMap` like that used in the hedera API to bytes of a transaction. 
+* `Submit` provides the requesting entity with a response containing information of a signed transaction. 
+* `SignAndSubmit` allows for both the sign and submit actions to be carried out in half of the number of responses and requests. 
+
+These calls allow for a single transaction to be signed by multiple keys for multi signature protocols.
+
+## Specification
+
+```
+message LimitsRequest{
+    required int32 MaxTransactionFee  = 1; //maybe we make these types bigger to prevent overflow
```

Comment:
Perhaps this type should match what a TX max transaction fee is for a `TransactionBody`, that would make it `uint64`?